### PR TITLE
List view pills

### PR DESCRIPTION
**Pull request by AnshG714**
## Summary

This PR adds the 'Pill Filters' as seen in the new Figma Designs, with some tweaks.
- Add Pills
- Add filtering mechanism
- Add Categories to relevant components in iOS and Android ListExample files. 

### Normal ListView
Simulator Screen Shot - iPhone 8 - 2020-07-21 at 17 16 36

==========================
### Pills persist appearance and position even when search is going on
Pill persistence

==========================
### Pure pill filtering (without search)
Pure pill filtering

==========================
### The search filtering works in conjunction with pill filtering!
search text + pills

### CODE REVIEW

File: packages/RNTester/js/components/RNTesterListFilters.js [hunk 1]
```diff
@@ -15,39 +15,55 @@ const React = require('react');
 const {StyleSheet, Text, TouchableOpacity, View} = require('react-native');
 const filters = ['Basic', 'UI', 'ListViews', 'iOS', 'Android'];
```
Comment (anku255) [inline packages/RNTester/js/components/RNTesterListFilters.js:16]: Adding a newline between imports and local variable declaration.

```suggestion

const filters = ['Basic', 'UI', 'ListViews', 'iOS', 'Android'];
```

**Comment by AnshG714** [comment]
> LGTM!
> 
> Just a nitpick regarding spacing of the pills. Currently, in my screen it looks like this -
> 
> Screenshot 2020-07-22 at 7 27 00 PM
> 
> We might wanna make it `justifyContent: 'space-around'` or increasing the `paddingHorizontal`. I experimented with `paddingHorizontal: 14` and it looks like this -
> 
> Screenshot 2020-07-22 at 7 26 41 PM

Thanks for the feedback Ankit! Yeah, I'll try to see what works best across both platforms + across different screen sizes as well.

**Comment by anku255** [comment]
@AnshG714 It has 3 approvals already. Should we merge it?

**Comment by manyaagarwal** [comment]
Since these filter options are only relevant to Components, should these not appear when the user has opened the APIs? Or be replaced with relevant filter pills for API? @AnshG714

**Comment by AnshG714** [comment]
@manyaagarwal this can be used for APIs as well.

**Comment by manyaagarwal** [comment]
@AnshG714 Similar categories do not exist for APIs and I am not sure how these categories would fit API filtering? Currently all of them have either been assigned "Other" or are platform specific.

**Pull request merged**